snuba/subscriptions/data.py

```diff
@@ -261,7 +265,9 @@ def validate_subscription(
             raise InvalidSubscriptionError("Only simple queries are supported")
         entity = get_entity(from_clause.key)
 
-        SubscriptionAllowedClausesValidator().validate(query)
+        SubscriptionAllowedClausesValidator(
+            self.entity_subscription.MAX_ALLOWED_AGGREGATIONS
+        ).validate(query)
```

Comment:
I think this would be more flexible if the validate function was actually on the entity subscription itself rather then just getting the MAX_ALLOWED_AGGREGATIONS from there and passing it to the SubscriptionAllowedClausesValidator. e.g. `self.entity_subscription.validate(query)`

It would allow us to add any kind of custom validations on an entity, not just enforce the maximum number of aggregations. It would allow to add much stricter validations on the subscription data and immediately reject subscriptions that don't meet that criteria rather than have them fail somewhere later in the pipeline.

Obviously this works fine now as the only validation we have is on max aggregations anyway so I'll leave it up to you whether you want to take this approach or not.

Reply:
I agree with you.. logically it makes sense to just move the `validate` to `EntitySubscription`